init.lua

```diff
@@ -1333,20 +1338,162 @@ function torchcraft:get_layer(layer_type, also_enemy)
+                img = self:draw_entity(img, pos_x, pos_y,
+                                       ut.pixel_size_x, ut.pixel_size_y, color)
+            end
+        end
     else
         for uid, ut in pairs(t) do
             if self:is_unit_in_screen(ut) then
                 pos_x, pos_y = self:get_unit_screen_pos(ut)
-                if layer_type == "hp" then
+                if layer_type == "hp" and ut.max_hp > 0 then
                     color = utils.get_health_color(ut.hp, ut.max_hp)
                 elseif layer_type == "type" then
                     color = utils.html_color_table[ut.type]
                 elseif layer_type == "player" then
                     color = utils.players_color_table[ut.playerId]
+                elseif layer_type == "shield" and ut.shield > 0 then
+                    color = utils.get_health_color(ut.shield, ut.max_shield)
+                elseif layer_type == "energy" then
+                    color = utils.get_health_color(ut.energy, 250)
                 end
-                img = self:draw_entity(img, pos_x, pos_y,
-                                       ut.pixel_size_x, ut.pixel_size_y,
-                                       color)
+                if color then
+                    img = self:draw_entity(img, pos_x, pos_y,
+                                           ut.pixel_size_x, ut.pixel_size_y,
+                                           color)
+                end
+            end
+        end
+    end
+    return img
+end
+
+function torchcraft:draw_value(img, pos_x, pos_y, size_x, size_y, value)
+    local narrow_width_start = math.max(1,
+                                        math.floor(pos_x - size_x / 2))
+    local narrow_width_end = math.min(self.field_size[1],
+                                      math.floor(pos_x + size_x / 2))
+    local narrow_height_start = math.max(1,
+                                         math.floor(pos_y - size_y / 2))
+    local narrow_height_end = math.min(self.field_size[2],
+                                       math.floor(pos_y + size_y / 2))
+    -- Hack to fix crash were unit is on top left corner of FoV
+    if narrow_width_end <= 0 then
+        narrow_width_end = narrow_width_start
+    end
+    local rect = img:narrow(2, narrow_width_start,
+                            narrow_width_end - narrow_width_start + 1)
+    if narrow_height_end <= 0 then
+        narrow_height_end = narrow_height_start
+    end
+    local rect = rect:narrow(1, narrow_height_start,
+                             narrow_height_end - narrow_height_start + 1)
+    rect:fill(value)
+    return img
+end
+
+function torchcraft:get_feature(feature, also_enemy, dtype)
+--returns 2D imagelike tensor of feature
+--type defaults to ByteTensor
+
+-- :TODO: Maybe make categorical features return a 3D tensor of one-hots
+--        Or just render them to colors with draw value to increase spacing
+
+    local img
+    if dtype == "short" then
```

Comment:
In Torch you can easily create different tensor types, so there is no need to have an additional argument specifying this. Also `img` is instantiated earlier and should be reused.